templates/tower-project.yaml

```diff
@@ -0,0 +1,454 @@
+    Description: S3 bucket objects will transition into this storage class
+    AllowedValues:
+      - DEEP_ARCHIVE
+      - INTELLIGENT_TIERING
+      - STANDARD_IA
+      - ONEZONE_IA
+      - GLACIER
+    Default: GLACIER
+
+  LifecycleDataExpiration:
+    Type: Number
+    Description: >-
+      Number of days (from creation) when objects are
+      deleted from S3 and the LifecycleDataStorageClass
+    Default: 365000
+
+  AdminArns:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role ARNs (strings) who should have access to
+      project-specific resources (e.g., private S3 bucket, KMS key, secrets).
+
+  NonAdminArns:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role ARNs (strings) who shouldn't have access to
+      project-specific resources (e.g., private S3 bucket, KMS key, secrets).
+
+  TemplateRootUrl:
+    Type: String
+    Description: URL of S3 bucket where templates are deployed
+    ConstraintDescription: Must be a valid S3 HTTP URL
+
+  TowerForgePolicyArn:
+    Type: String
+    Description: ARN for Tower Forge IAM policy
+
+  TowerLaunchPolicyArn:
+    Type: String
+    Description: ARN for Tower Launch IAM policy
+
+Conditions:
+  HasReadWriteAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], ""]]
+  HasReadOnlyAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], ""]]
+  HasAnyAccessArns:
+    !Or [!Condition HasReadWriteAccessArns, !Condition HasReadOnlyAccessArns]
+
+Resources:
+  TowerForgeServiceUser:
+    Type: AWS::IAM::User
+
+  TowerForgeServiceUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      UserName: !Ref TowerForgeServiceUser
+
+  TowerForgeServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      ManagedPolicyArns:
+        - !Ref TowerForgePolicyArn
+        - !Ref TowerLaunchPolicyArn
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS:
+                - !GetAtt TowerForgeServiceUser.Arn
+              Service:
+                - ecs.amazonaws.com
+                - batch.amazonaws.com
+            Action:
+              - sts:AssumeRole
+
+  TowerForgeInstanceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      ManagedPolicyArns:
+        - "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess"
+        - "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+                - ecs.amazonaws.com
+                - batch.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: "ForgeBatchJobPolicy"
+          PolicyDocument: 
+            Version: "2012-10-17"
+            Statement:
+              - Sid: ForgeInstanceRoleBatchJob
```

Comment:
This appears to largely be a duplication of an existing stack, one of the first created in the account,
`nextflow-forge-iam-policy`. I _generally_ find it preferable to use managed policies rather than inline ones, and you can making modifications to the existing one if necessary, since the original rollout of two required policies and a service user is being updated by this template.

Reply:
Sorry, I pushed this right before logging off so that I could link to the template I was testing for Seqera. If this ends up being necessary, most of it will be moved to a managed policy. The only inline policy will be for resource-specific statements (like the KMS key).

Reply:
I updated the link I gave to Seqera to a commit permalin. I removed the InstanceRole from the template and disabled bucket encryption until Seqera offers us a workaround (so that I can test things in the meantime).